builder/processor/src/main/java/io/helidon/builder/processor/BlueprintProcessor.java

```diff
@@ -253,331 +259,223 @@ private void generatePrototypeWithBuilder(TypeElement builderInterface,
-                if (!generated.javadoc().isEmpty()) {
-                    pw.print(SOURCE_SPACING);
-                    pw.println("/**");
-                    for (String docLine : generated.javadoc()) {
-                        pw.print(SOURCE_SPACING);
-                        pw.print(" *");
-                        pw.println(docLine);
-                    }
-                    pw.print(SOURCE_SPACING);
-                    pw.println(" */");
-                }
-                for (String annotation : customMethod.generatedMethod().annotations()) {
-                    pw.print(SOURCE_SPACING);
-                    pw.print('@');
-                    pw.println(annotation);
-                }
-                pw.print(SOURCE_SPACING);
-                pw.print("static ");
-                pw.print(generated.returnType().fqName());
-                pw.print(" ");
-                pw.print(generated.name());
-                pw.print("(");
-                pw.print(generated.arguments()
-                                 .stream()
-                                 .map(it -> it.typeName().fqName() + " " + it.name())
-                                 .collect(Collectors.joining(", ")));
-                pw.println(") {");
-                pw.print(SOURCE_SPACING);
-                pw.print(SOURCE_SPACING);
-                pw.print(customMethod.generatedMethod().callCode());
-                pw.println(";");
-                pw.print(SOURCE_SPACING);
-                pw.println("}");
-                pw.println();
-            }
+        generateCustomMethods(customMethods, classModel);
+
+        /*
+          abstract class BuilderBase...
+         */
+        GenerateAbstractBuilder.generate(classModel,
+                                         typeInformation.prototype(),
+                                         typeInformation.runtimeObject().orElseGet(typeInformation::prototype),
+                                         typeArguments,
+                                         typeContext);
+        /*
+          class Builder extends BuilderBase ...
+         */
+        GenerateBuilder.generate(classModel,
+                                 typeInformation.prototype(),
+                                 typeInformation.runtimeObject().orElseGet(typeInformation::prototype),
+                                 typeArguments,
+                                 typeContext.blueprintData().isFactory(),
+                                 typeContext);
 
-            for (CustomMethods.CustomMethod customMethod : customMethods.prototypeMethods()) {
-                // prototype definition - custom methods must have a new method defined on this interface, missing on blueprint
-                CustomMethods.Method generated = customMethod.generatedMethod().method();
+        try (PrintWriter pw = new PrintWriter(generatedIface.openWriter())) {
+            classModel.build()
+                    .saveToFile(pw, SOURCE_SPACING);
```

Comment:
`saveToFile()` seems like a strange choice of name here, shouldn't it be just `write()` instead?